rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -124,6 +143,45 @@ rocalTensorInfo::rocalTensorInfo(std::vector<size_t> dims,
     if (_num_of_dims <= 3) _is_image = false;
 }
 
+rocalTensorInfo::rocalTensorInfo(const rocalTensorInfo &other) {
+    _type = other._type;
+    _num_of_dims = other._num_of_dims;
+    _dims = other._dims;
+    _batch_size = other._batch_size;
+    _mem_type = other._mem_type;
+    _roi_type = other._roi_type;
+    _data_type = other._data_type;
+    _layout = other._layout;
+    _color_format = other._color_format;
+    _data_type_size = other._data_type_size;
+    _data_size = other._data_size;
+    _max_shape = other._max_shape;
+    _is_image = other._is_image;
+    _is_metadata = other._is_metadata;
+    _channels = other._channels;
+    if(!other.is_metadata()) {  // For Metadata ROI buffer is not required
+        allocate_tensor_roi_buffers();
+        memcpy((void *)_roi_buf, (const void *)other.get_roi(), _batch_size * 4 * sizeof(unsigned));
+    }
+}
+
+rocalTensorInfo::~rocalTensorInfo() {
+    if(!_is_metadata) {
+        if(_mem_type == RocalMemType::HIP) {
+#if ENABLE_HIP
```

Comment:
OPENCL is not handled here

Reply:
We will create a separate PR for OpenCL changes